File: TaskManagmentBackEnd/TaskManagement/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManagement.Models;

public class Task : Entity
{
    private const int GUID_LENGTH = 36;
    
    [MaxLength(20)]
    public string Title { get; set; } = null!;

    [MaxLength(200)]
    public string? Description { get; set; }
    
    public DateTime DueDate { get; set; }
    
    public Status Status { get; set; } = Status.Pending;
    public User User { get; set; } = null!;
    
    [MaxLength(GUID_LENGTH)]
    public string UserId { get; set; } = null!;
}

public enum Status
{
    Pending = 0,
    Complete = 1,
}

File: TaskManagmentFrontEnd/src/app/Models/Task.ts
export class Task {
    id: number = 0;
    title: string = '';
    description: string = '';
    dueDate: Date = new Date();
    status: Status = Status.pending;
    assignee: string = '';
    assigneeId: string = '';
}



export enum Status{
    pending,
    completed
}